File: 1967-number-of-strings-that-appear-as-substrings-in-word/solution.cpp
#include <vector>
#include <string>

class Solution {
public:
    int numOfStrings(std::vector<std::string>& patterns, std::string word) {
        int count = 0;
        
        // Iterate through each pattern in the patterns vector
        for (const std::string& pattern : patterns) {
            // Check if the current pattern exists as a substring in the word.
            // string::npos is returned if the substring is not found.
            if (word.find(pattern) != std::string::npos) {
                count++;
            }
        }
        
        return count;
    }
};

File: 1967-number-of-strings-that-appear-as-substrings-in-word/solution.py
class Solution:
    def numOfStrings(self, patterns: list[str], word: str) -> int:
        count = 0
        
        # Iterate through each pattern in the list
        for pattern in patterns:
            # The 'in' operator in Python efficiently checks 
            # if a substring exists within a string.
            if pattern in word:
                count += 1
                
        return count